static/app/stores/sidebarProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import {createContext, useContext, useReducer} from 'react';
+
+function assertNever(x: never) {
```

Comment:
Why the separate function? Why not just throw the type error in the default case?

Reply:
This does the exhaustive check at type checking and we avoid throwing in prod